src/main/java/htsjdk/samtools/TextTagCodec.java

```diff
@@ -98,8 +98,11 @@ private char getArrayType(final Object array, final boolean isUnsigned) {
     }
 
     private String encodeArrayValue(final Object value) {
-        final StringBuilder ret = new StringBuilder(Array.get(value, 0).toString());
         final int length = Array.getLength(value);
+        if(length == 0){
```

Comment:
Testing the new "add suggestion feature"

```suggestion
        if (length == 0){
```